### PR TITLE
Added raster geopackage checker

### PR DESCRIPTION
**Pull request by ywkim312**
This PR will check if the geopackage file contains raster layer or no vector date layer.
To test, please use attached geopackage file

Scenario1: Uploading the raster based geopackage file
1. download attached geopackages.zip file and unzip it. Should be two files, one named guid_test.gpkg, and the other is landcover.gpkg
2. create a dataset using the following json (the content doesn't matter too much because it is test. Only important one is the format, that should be geopackage)
{"format": "geopackage", "title": "test-delete", "description": "delete me", "dataType": "incore:addressPoints"}
3. attached the geopackage file to the dataset using the landcover.gpkg file (raster geopackage)
4. if you use landocver.gpkg, then you should see the error message saying, it doesn't contain any vector layer or it contains the raster
5. if you get error in step 3 above, try to attach the guid_test.gpkg (vector data). This should work without any error.
6. Remove the dataset you created in step 1.
[geopackages.zip](https://github.com/IN-CORE/incore-services/files/14029143/geopackages.zip)

### CODE REVIEW

File: server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeotoolsUtils.java [hunk 1]
```diff
@@ -969,33 +970,47 @@ public static boolean doesGuidExist(SimpleFeatureCollection inputFeatures) {
...
                 String fileName = inFile.getName().split("\\.")[0];
-                if (layerName.equals(fileName)) {
-                    output = true;
+                if (!layerName.equals(fileName)) {
+                     return 3;
                 }
+            } else if (layerNames.length == 0) {
+                return 1;
+            } else if (layerNames.length > 1) {
+                return 2;
```
Comment (longshuicy) [inline server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeotoolsUtils.java:1013]: Is using 1, 2, 3 a good pattern? Could it be more specific?

Comment (ylyangtw) [reply server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeotoolsUtils.java:1013]: Maybe something like this? 
```
public enum gpkgValidationResult {
    VALID, 
    RASTER_OR_NO_VECTOR_LAYER, 
    MULTIPLE_VECTOR_LAYERS, 
    NAME_MISMATCH
}

```

**Review by ylyangtw — APPROVED** [review]
Tested the scenarios. All worked as expected. Changes look good. Approving~

File: server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeotoolsUtils.java (hunk 1 above)
Comment (ywkim312) [reply server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeotoolsUtils.java:1013]: I have updated based on Ya-lan's suggestion. Thank you @ylyangtw

**Review by longshuicy — APPROVED** [review]

**Review by longshuicy — APPROVED** [review]

**Pull request merged**